src/Sylius/Bundle/ApiBundle/Converter/PayumReplyConverter.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Converter;
+
+use LogicException;
+use Payum\Core\Bridge\Symfony\Reply\HttpResponse as SymfonyHttpResponse;
+use Payum\Core\Reply\HttpResponse;
+use Payum\Core\Reply\ReplyInterface;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+/**
+ * In the Payum world, The ReplyInterface is used by :
+ * - Payum\Core\Reply\HttpResponse
+ * - Payum\Core\Bridge\Symfony\Reply\HttpResponse
+ *
+ * If for some reason a Payum gateway is using another kind of class then this service
+ * must be decorated by this gateway to ensure that the ReplyInterface object will
+ * be converted to the right JSON array
+ *
+ * @experimental
+ */
+final class PayumReplyConverter implements PayumReplyConverterInterface
+{
+    public function convert(ReplyInterface $reply): JsonResponse
+    {
+        if ($reply instanceof SymfonyHttpResponse) {
+            $response = $reply->getResponse();
+
+            $content = $response->getContent();
+            /** @var string[] $headers */
+            $headers = $response->headers->all();
+
+            $reply = new HttpResponse(
+                $content === false ? '' : $content,
+                $response->getStatusCode(),
+                $headers,
+            );
+        }
+
+        if ($reply instanceof HttpResponse) {
+            return new JsonResponse([
+                'content' => $reply->getContent(),
+                'statusCode' => $reply->getStatusCode(),
+                'headers' => $reply->getHeaders(),
```

Comment:
I think it would be nice to also provide the contents of the Location header, if present, as its own array element. This seems to be a very common scenario, so why not make it easier to implement.

Reply:
Location header is a header so all is inside the `headers` `array` 😉

Reply:
I know, what I suggested is VIP treatment for it. :)

Reply:
I agree adding something specific only for this case could be done, but if you think about all possible cases then this converter have to remain generic, if someone need the `Location` header to be explicitly displayed then he just have to decorate this service.